### PR TITLE
feat: [STO-4648]: use STO token endpoint v2

### PR DESCRIPTION
**Pull request by ahokinson**
## Describe your changes
Migrate from v1 to v2 of the sto-core endpoints. Token responses have changed to a JSON response instead of a string response. Handling them, we now parse the response for JSON and return only the token.

## Comment Triggers
<details>
  <summary>Build triggers</summary>
  
- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>
  
  You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`
  
- Compile: `trigger compile`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/37224)

### CODE REVIEW

**Comment by diptiman-adak** [comment]
Quality gate Quality gate: **_Passed_** 
 


            
 No issues Everything looks good
 [Details](https://app.embold.io/organization/gh/harness/repositories/7752dbb3099d2dba3b5d37f5931a5e4c/pull-request/37224)

File: 323-sto-utilities/src/main/java/io/harness/stoserviceclient/STOServiceUtils.java [hunk 1]
```diff
@@ -61,6 +63,9 @@ public String getSTOServiceToken(String accountID) {
           "Could not fetch token from STO service. status code = %s, message = %s, response = %s", response.code(),
           response.message() == null ? "null" : response.message(), response.errorBody() == null ? "null" : errorBody));
     }
-    return response.body();
+
+    JsonObject tokenResponse = new JsonParser().parse(response.body()).getAsJsonObject();
```
Comment (r0610205) [inline 323-sto-utilities/src/main/java/io/harness/stoserviceclient/STOServiceUtils.java:67]: This should show deprecated warning. Use `JsonParser.parseString()` instead to get rid of deprecated syntax

Comment (ahokinson) [reply 323-sto-utilities/src/main/java/io/harness/stoserviceclient/STOServiceUtils.java]: [I see what you mean](https://javadoc.io/doc/com.google.code.gson/gson/latest/com.google.gson/com/google/gson/JsonParser.html). I borrowed this format from elsewhere in the project.

**Review by r0610205 — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/37224)* status: :shipit: complete! ✓

**Pull request merged**